DetectorDescription/DDCMS/src/DDFilteredView.cc

```diff
@@ -12,20 +12,17 @@ using namespace edm;
 using namespace std;
 using namespace cms::dd;
 
-DDFilteredView::DDFilteredView(const DDDetector* det, const Volume volume)
-    : node_(volume->GetNode(0)), registry_(&det->specpars()) {
+DDFilteredView::DDFilteredView(const DDDetector* det, const Volume volume) : registry_(&det->specpars()) {
```

Comment:
Did you really want to remove initialization of `node_`?

Reply:
not really, I'm not sure this would be a correct node though. I'd rather leave it uninitialised then initialised to a wrong one. I'm writing more unit tests to make sure the assignment and navigation are indeed what's expected.

Reply:
I was thinking that `node_ != nullptr` was an invariant of `DDFilteredView`, but apparently it is not.
I count seven `assert(node_);` statements in the `DDFilteredView` methods. Any slight error in the sequence of calls to `DDFilteredView` methods could result in failed asserts. Maybe the constructor should perform some initialization so the object starts out in a usable state?

Reply:
@cvuosalo do you have an explicit suggestion? As it is now it point to ```nullptr```, anyway having many ```assert``` statements should just help ensuring code robustness, no?

Reply:
It's a bit worrisome that after constructing a `DDFilteredView` for reading, the `next()` call is mandatory. `next()` is, in effect, part of the construction process. We will have to be careful to ensure there is no path in the code where a `DDFilteredView` is constructed and then get its methods called without `next()` first being called, or else failed asserts will occur.

Reply:
```DDCompactView``` is going to define a rootNode (aka ``` rootNodeName = cms.string('cms:OCMS')```) that will be propagated to the ```DDFilteredView``` created from it.